# treedater-earlySL

This is a reanalysis of Ebola virus sequences originally presented by [Gire et al.](http://dx.doi.org/10.1126/science.1259657) and later reanalysied by [Moller et al.](https://doi.org/10.1073/pnas.1713314115).
Moller et al. suggest that substitution rate estimates based on these data are very sensitive to the population genetic prior used in Bayesian analyses. 
We're therefore curious what we estimate when we use a pseudo-maximum likelihood approach that does not incorporate such a prior.

## treedater 
The _treedater_ algorithm is [described here](https://doi.org/10.1093/ve/vex025). This is an [R package](https://github.com/emvolz/treedater), and we can very quickly estimate rates and dates from the 81 whole genomes used by Gire and Moller et al. 

Load the packages and data: 
```{r}
require(treedater)
require(ape)

( aln <- read.dna( 'gire_coding.fas', 'fasta') )
```

We will do a *very lazy* phylogenetic analysis. Afterall, it's Sunday and we need to watch _Lost in Space_. 
Following Moller, we will use a HKY model with no rate variation between sites. Then we will estimate a NJ tree. 
```{r}
D <- dist.dna( aln, model = 'F84', pairwise.del = TRUE )
tre <- unroot( bionj( D ) )
```


To estimate a molecular clock, we must extract the date from the sequence labels. These take the form of (for example) "EBOV|AA000000|EM104|SierraLeone_EM|2014-06-02"
```{r}
require(magrittr)
# note: times will be in units of years after Jan 1, 2014
sts <- strsplit( tre$tip.label, '\\|') %>% sapply( function(x) tail(x,1) ) %>% as.Date %>% -as.Date('2014-01-01' )/365
# note: making this a named vector corresponding to tree tip labels 
sts <- sts %>% as.numeric %>% setNames( tre$tip.label )
```

Now we can run _treedater_ with a relaxed clock
```{r}
( dtr.relaxed <- dater( tre, sts , s = 14e3 ) )
```
And parametric bootstrap for the confidence intervals
```{r}
(boot.dtr.relaxed <-  parboot.treedater( dtr.relaxed , ncpu = 8)  )
```

And likewise with a strict clock, which is more like what was done in Moller et al
```{r}
( dtr.strict <- dater( tre, sts , s = 14e3 , strict=TRUE) )
(boot.dtr.strict <-  parboot.treedater( dtr.strict , ncpu = 8)  )
```

## Summary 
* Our rates are a little lower, but close to, what was reported by Moller et al. using the structured coalescent prior, especially with a strict clock.
* The estimated TMRCA is in early 2014, which is an acceptable estimate. This is intermediate between the estimates using structured coalescent and other priors reported by Muller. 
* We can't very well estimate the lower bound of TMRCA from this small aligment using the parametric bootstrap. The priors used in the Bayesian methods probably have a large influence on that. 
* Estimates could not doubt be improved by doing a proper phylogenetic analysis and fine-tuning parameters
* All of the preceding analysis took less than one minute of computation time. 
